src/org/labkey/test/tests/query/QueryMetadataTest.java

```diff
@@ -0,0 +1,210 @@
+package org.labkey.test.tests.query;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.labkey.test.BaseWebDriverTest;
+import org.labkey.test.pages.query.EditMetadataPage;
+import org.labkey.test.params.FieldDefinition;
+import org.labkey.test.params.experiment.SampleTypeDefinition;
+import org.labkey.test.params.list.IntListDefinition;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+
+@Category({})
+public class QueryMetadataTest extends BaseWebDriverTest
+{
+    static public String TEST_LIST = "queryMetadataTestList";
+    static public String TEST_SAMPLES = "queryMetadataSamples";
+
+    @Override
+    protected void doCleanup(boolean afterTest)
+    {
+        _containerHelper.deleteProject(getProjectName(), afterTest);
+    }
+
+    @BeforeClass
+    public static void setupProject() throws Exception
+    {
+        QueryMetadataTest init = (QueryMetadataTest) getCurrentTest();
+
+        init.doSetup();
+    }
+
+    private void doSetup() throws Exception
+    {
+        _containerHelper.createProject(getProjectName(), "Collaboration");
+
+        // create a sampleType to look up to
+        List<FieldDefinition> fields = List.of(
+                new FieldDefinition("stringField", FieldDefinition.ColumnType.String),
+        new FieldDefinition("boolField", FieldDefinition.ColumnType.Boolean),
+        new FieldDefinition("decimalField", FieldDefinition.ColumnType.Decimal));
+        var samplesDgen = new SampleTypeDefinition(TEST_SAMPLES)
+                .setFields(fields)
+                .create(createDefaultConnection(), getProjectName())
+                .withGeneratedRows(10);
+        samplesDgen.insertRows();
+        // create a list
+
+        List<FieldDefinition> listColumns = Arrays.asList(
+                new FieldDefinition("name", FieldDefinition.ColumnType.String),
+                new FieldDefinition("value", FieldDefinition.ColumnType.Integer),
+                new FieldDefinition("users", FieldDefinition.ColumnType.User),
+                new FieldDefinition("date", FieldDefinition.ColumnType.DateAndTime),
+                new FieldDefinition("samples", new FieldDefinition.IntLookup(getProjectName(),"exp.materials", TEST_SAMPLES)),
+                new FieldDefinition("selfLookup", new FieldDefinition.IntLookup(getProjectName(),"lists", TEST_LIST)));
+        var dgen = new IntListDefinition(TEST_LIST, "Key").setFields(listColumns)
+                .create(createDefaultConnection(), getProjectName());
+    }
+
+    @Test
+    public void testUpdateLookupFields()
```

Comment:
It would be helpful for these test cases to link out to specific Issues they're attempting to cover for regressions. Always find those useful.